### PR TITLE
DEP: integrate.quad_vec: deprecate parameter full_output

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

https://github.com/scipy/scipy/pull/17837#issuecomment-1403785816
#### What does this implement/fix?

Deprecates the parameter `full_output`

### CODE REVIEW

File: scipy/integrate/_quad_vec.py
Comment (j-bowhay) [inline scipy/integrate/_quad_vec.py:162]:
```suggestion
           are provided as attributes of the result object.
```

File: scipy/integrate/_quad_vec.py
Comment (j-bowhay) [inline scipy/integrate/_quad_vec.py:158]: Was actually a bunch not a dict

File: scipy/integrate/_quad_vec.py
Comment (j-bowhay) [inline scipy/integrate/_quad_vec.py:191]: This was undocumented previously

File: scipy/integrate/_quad_vec.py
Comment (j-bowhay) [inline scipy/integrate/_quad_vec.py:244]:
```suggestion
               "accessed as attributes of the object returned by "
```

File: scipy/integrate/_quad_vec.py
Comment (mdhaber) [inline scipy/integrate/_quad_vec.py:455]: I'm not sure this would have passed the PEP8 checks before all the changes. (It seems to be a lot more relaxed now...) It is not aligned with the `(` above, and it is over-indented otherwise.

File: scipy/integrate/_quad_vec.py
Comment (mdhaber) [inline scipy/integrate/_quad_vec.py:172]: The object was just a tuple before, so I think it's reasonable to re-think the names. I called this first quantity `integral` in `nquad`, which I think is more descriptive than `res`. Whether `err` should be `abserr` as in `nquad` or if we should change it to `error` across the board is debatable. The rest of the names look OK to me.

File: scipy/integrate/_quad_vec.py
Comment (mdhaber) [inline scipy/integrate/_quad_vec.py:444]: Let's add a note in the deprecation tracker than when `full_output` is removed, `_Bunch` can be removed.

**Review by mdhaber — COMMENTED** [review]
This looks pretty close; I'd just reconsider some of the attribute names while you're fixing the refguide failure. 
I tested it locally and the warnings are emitted as intended when `full_output` is specified. When `full_output` is False or not specified, we get the bunch object; when `full_output=True`, we still get the tuple (during the deprecation period). Looks like all is as it should be.

File: scipy/integrate/_quad_vec.py
Comment (j-bowhay) [inline scipy/integrate/_quad_vec.py:245]:
```suggestion
               "'scipy.integrate.quad_vec'.")
```

**Comment by j-bowhay** [comment]
Thanks for taking a look @mdhaber, hopefully have addressed your comments

File: scipy/integrate/tests/test__quad_vec.py
Comment (mdhaber) [inline scipy/integrate/tests/test__quad_vec.py:231]: I wouldn't mind using `assert_equal` throughout for consistency, but this isn't enough to re-run CI.

**Review by mdhaber — APPROVED** [review]
There's a minor note in the documentation and tests, but I'd say just fix that in the `quad` PR. ~~I'll go ahead and merge this. Let's have someone else (@tupui?) review the `quad` PR so that effectively, any decisions we made here get another set of eyes on them.~~ @j-bowhay actually I'd appreciate your thoughts on https://github.com/scipy/scipy/pull/17964#discussion_r1103890157 before we merge.

File: scipy/integrate/_quad_vec.py
Comment (mdhaber) [inline scipy/integrate/_quad_vec.py:175]:
```suggestion
        integral : {float, array-like}
            Estimate of the integral.
        abserr : float
            Norm of the individual integral error estimates (as specified by
            `norm`).
```
We have the chance now to reconsider whether we should return each of the errors individually. Same goes for `success`, `status`, maybe `neval`, etc. Personally, I think that if we're going to vectorize functions like this, they should provide essentially the same information as if we were looping over the scalar version. For example, see what I'm proposing for `chandrupatla` (a scalar root finder) in https://github.com/scipy/scipy/issues/7242#issuecomment-1370131973.

Comment (j-bowhay) [reply scipy/integrate/_quad_vec.py:175]: > We have the chance now to reconsider whether we should return each of the errors individually.

Is this not what `errors` contains?

Looking at the code I think it would be feasible for the others. I have not used this function in anger so don't have a strong opinion on how useful this would be.

Comment (mdhaber) [reply scipy/integrate/_quad_vec.py:175]: Oops, I didn't check. It's documented as "`float`".

Comment (j-bowhay) [reply scipy/integrate/_quad_vec.py:175]: You're right I am confusing the intervals vs components of the vectorised function

Comment (j-bowhay) [reply scipy/integrate/_quad_vec.py:175]: I had another look over the code; the part where the norm of taken is about as deep into the internals of the code as you can get. Whilst it probably would be possible to do what your suggesting it would be a non-trivial change so perhaps beyond the scope of this pr?

Comment (mdhaber) [reply scipy/integrate/_quad_vec.py:175]: Ok, no problem! We'll leave it alone.

**Comment by mdhaber** [comment]
Thanks @j-bowhay!

**Comment by j-bowhay** [comment]
Forgot to say, @mdhaber would you mind adding this to the tracker and release notes :)

**Comment by mdhaber** [comment]
Done.

**Pull request merged**
